spyder/plugins/editor/widgets/base.py

```diff
@@ -187,26 +188,30 @@ def update_list(self, filter_text):
                      CompletionItemKind.KEYWORD: 'keyword',
                      CompletionItemKind.TEXT: 'text'}
 
-        for completion in self.completion_list:
+        self.display_index = []
+
+        for i, completion in enumerate(self.completion_list):
             if not self.is_internal_console:
                 completion_label = completion['filterText']
                 icon = icons_map.get(completion['kind'], 'no_match')
                 item = QListWidgetItem(ima.icon(icon),
-                                       completion['insertText'])
+                                       completion['label'])
             else:
                 completion_label = completion[0]
                 item = QListWidgetItem(completion_label)
 
             if self.check_can_complete(
                     completion_label, filter_text):
                 self.addItem(item)
+                self.display_index.append(i)
 
         if self.count() > 0:
             self.setCurrentRow(0)
             self.scrollTo(self.currentIndex(),
                           QAbstractItemView.PositionAtTop)
         else:
-            self.addItem(QListWidgetItem(self.empty_text))
+            # self.addItem(QListWidgetItem(self.empty_text))

```

Comment:
This fixes also #10002

Reply:
I'd prefer if you do it in a separate PR. I know it's very simple, but I'd like to merge it right away.